File: server/src/main/kotlin/com/notebox/domain/reminder/Reminder.kt
package com.notebox.domain.reminder

import com.notebox.dto.ReminderDto
import java.time.Instant

data class Reminder(
    val id: String,
    val noteId: String,
    val userId: String,
    val title: String,
    val remindAt: Instant,
    val repeatType: RepeatType,
    val repeatEndAt: Instant?,
    val notificationSent: Boolean,
    val googleEventId: String?,
    val createdAt: Instant,
    val updatedAt: Instant
) {
    fun toDto() = ReminderDto(
        id = id,
        noteId = noteId,
        title = title,
        remindAt = remindAt.toEpochMilli(),
        repeatType = repeatType.name,
        repeatEndAt = repeatEndAt?.toEpochMilli(),
        notificationSent = notificationSent,
        googleEventId = googleEventId,
        createdAt = createdAt.toEpochMilli(),
        updatedAt = updatedAt.toEpochMilli()
    )
}

enum class RepeatType {
    NONE, DAILY, WEEKLY, MONTHLY, YEARLY
}

File: src/types/reminder.ts
export type RepeatType = 'NONE' | 'DAILY' | 'WEEKLY' | 'MONTHLY' | 'YEARLY';

export interface Reminder {
  id: string;
  noteId: string;
  title: string;
  remindAt: number; // timestamp
  repeatType: RepeatType;
  repeatEndAt?: number;
  notificationSent: boolean;
  googleEventId?: string;
  createdAt: number;
  updatedAt: number;
}

export interface CreateReminderRequest {
  noteId: string;
  title?: string;
  remindAt: number;
  repeatType?: RepeatType;
  repeatEndAt?: number;
  syncToGoogleCalendar?: boolean;
}

export interface UpdateReminderRequest {
  title?: string;
  remindAt?: number;
  repeatType?: RepeatType;
  repeatEndAt?: number;
}

export interface PushSubscription {
  id: string;
  userId: string;
  endpoint: string;
  createdAt: number;
}

export interface SubscribePushRequest {
  endpoint: string;
  p256dh: string;
  auth: string;
}

export interface UnsubscribePushRequest {
  endpoint: string;
}
